app/models/complaint.rb

```diff
@@ -0,0 +1,119 @@
+    cobfc: /other/
+  }.freeze
+
+  FAC_CODE_ROLL_UP_SUMS = {
+    complaints_facility_code: :cfc,
+    complaints_financial_by_fac_code: :cfbfc,
+    complaints_quality_by_fac_code: :cqbfc,
+    complaints_refund_by_fac_code: :crbfc,
+    complaints_marketing_by_fac_code: :cmbfc,
+    complaints_accreditation_by_fac_code: :cabfc,
+    complaints_degree_requirements_by_fac_code: :cdrbfc,
+    complaints_student_loans_by_fac_code: :cslbfc,
+    complaints_grades_by_fac_code: :cgbfc,
+    complaints_credit_transfer_by_fac_code: :cctbfc,
+    complaints_job_by_fac_code: :cjbfc,
+    complaints_transcript_by_fac_code: :ctbfc,
+    complaints_other_by_fac_code: :cobfc
+  }.freeze
+
+  OPE6_ROLL_UP_SUMS = {
+    complaints_main_campus_roll_up: :cfc,
+    complaints_financial_by_ope_id_do_not_sum: :cfbfc,
+    complaints_quality_by_ope_id_do_not_sum: :cqbfc,
+    complaints_refund_by_ope_id_do_not_sum: :crbfc,
+    complaints_marketing_by_ope_id_do_not_sum: :cmbfc,
+    complaints_accreditation_by_ope_id_do_not_sum: :cabfc,
+    complaints_degree_requirements_by_ope_id_do_not_sum: :cdrbfc,
+    complaints_student_loans_by_ope_id_do_not_sum: :cslbfc,
+    complaints_grades_by_ope_id_do_not_sum: :cgbfc,
+    complaints_credit_transfer_by_ope_id_do_not_sum: :cctbfc,
+    complaints_jobs_by_ope_id_do_not_sum: :cjbfc,
+    complaints_transcript_by_ope_id_do_not_sum: :ctbfc,
+    complaints_other_by_ope_id_do_not_sum: :cobfc
+  }.freeze
+
+  MAP = {
+    'case id' => { column: :case_id, converter: BaseConverter },
+    'level' => { column: :level, converter: BaseConverter },
+    'status' => { column: :status, converter: BaseConverter },
+    'case owner' => { column: :case_owner, converter: BaseConverter },
+    'school' => { column: :institution, converter: InstitutionConverter },
+    'opeid' => { column: :ope, converter: OpeConverter },
+    'facility code' => { column: :facility_code, converter: FacilityCodeConverter },
+    'school city' => { column: :city, converter: BaseConverter },
+    'school state' => { column: :state, converter: BaseConverter },
+    'submitted' => { column: :submitted, converter: BaseConverter },
+    'closed' => { column: :closed, converter: BaseConverter },
+    'closed reason' => { column: :closed_reason, converter: BaseConverter },
+    'issues' => { column: :issues, converter: BaseConverter },
+    'education benefits' => { column: :education_benefits, converter: BaseConverter }
+  }.freeze
+
+  validates :facility_code, presence: true
+  validates :status, inclusion: { in: STATUSES }
+  validates :closed_reason, inclusion: { in: CLOSED_REASONS }, allow_blank: true
+
+  before_validation :derive_dependent_columns
+
+  def self.update_ope_from_crosswalk
+    find_each do |complaint|
+      crosswalk = Crosswalk.find_by(facility_code: complaint.facility_code)
+      complaint.update(ope: crosswalk.ope, ope6: crosswalk.ope6) if crosswalk
+    end
+  end
```

Comment:
AREL allows you to write queries in a way that is similar to raw SQL. It is bad practice to write something like Database.connection.execute(raw_sql)

With AREL you are still writing ruby vs sending a raw SQL statement. 

It's bad practice for a number of reasons, but most notably, there isn't a mechanism to prevent against things like SQL injection automatically. Raw sql is dangerous. But equally as important, the errors rendered with raw sql when a query fails are not as good as the ones when AREL query fails.